src/adc.c

```diff
@@ -70,6 +79,6 @@ static void CallCallback(struct adc_channel * channel_struct) {
 	if(channel_struct->ptr) {
 		channel_struct->callback(channel_struct->value, channel_struct->ptr);
 	}else {
-		channel_struct->callback(channel_struct->value);
+		channel_struct->callback(channel_struct->value, 0);
```

Comment:
Change this to cast the function pointer to the desired type instead of passing the null pointer to a function that presumably isn't expecting it:
(void(*)(uint16_t)channel_struct->callback(channel_struct->value);